lib/pages/main/home_page.dart

```diff
@@ -0,0 +1,427 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:app_tracking_transparency/app_tracking_transparency.dart';
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:mosquito_alert_app/api/api.dart';
+import 'package:mosquito_alert_app/models/notification.dart';
+import 'package:mosquito_alert_app/pages/forms_pages/adult_report_page.dart';
+import 'package:mosquito_alert_app/pages/forms_pages/biting_report_page.dart';
+import 'package:mosquito_alert_app/pages/forms_pages/breeding_report_page.dart';
+import 'package:mosquito_alert_app/pages/info_pages/info_page.dart';
+import 'package:mosquito_alert_app/pages/main/components/custom_card_widget.dart';
+import 'package:mosquito_alert_app/pages/main/components/navigation_drawer_widget.dart';
+import 'package:mosquito_alert_app/pages/my_reports_pages/my_reports_page.dart';
+import 'package:mosquito_alert_app/utils/MyLocalizations.dart';
+import 'package:mosquito_alert_app/utils/UserManager.dart';
+import 'package:mosquito_alert_app/utils/Utils.dart';
+import 'package:mosquito_alert_app/utils/style.dart';
+import 'package:mosquito_alert_app/utils/version_control.dart';
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  String? userName;
+
+  StreamController<String?> nameStream = StreamController<String?>.broadcast();
+  String? userUuid;
+  StreamController<bool> loadingStream = StreamController<bool>.broadcast();
+  int unreadNotifications = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    initAuthStatus();
+    //_getNotificationCount();
+  }
+
+  @override
+  void dispose() {
+    nameStream.close();
+    loadingStream.close();
+    super.dispose();
+  }
+
+  void initAuthStatus() async {
+    loadingStream.add(true);
+
+    if (Platform.isIOS) {
+      await AppTrackingTransparency.requestTrackingAuthorization();
+    }
+    VersionControl.getInstance().packageApiKey =
+        'uqFb4yrdZCPFXsvXrJHBbJg5B5TqvSCYmxR7aPuN2uCcCKyu9FDVWettvbtNV9HKm';
+    VersionControl.getInstance().packageLanguageCode = 'es';
+    var check = await VersionControl.getInstance().checkVersion(context);
+    if (check != null && check) {
+      _getData();
+    }
+
+    loadingStream.add(false);
+  }
+
+  void _getData() async {
+    await UserManager.startFirstTime(context);
+    userUuid = await UserManager.getUUID();
+    UserManager.userScore = await ApiSingleton().getUserScores();
+    await UserManager.setUserScores(UserManager.userScore);
+    await Utils.loadFirebase();
+
+    await Utils.getLocation(context);
+    if (UserManager.user != null) {
+      nameStream.add(UserManager.user!.email);
+      setState(() {
+        userName = UserManager.user!.email;
+      });
+    }
+  }
```

Comment:
shouldn't this be in main_vc.dart? it's more about initializing the app, not related exclusively to the home page

Reply:
Yes, I've moved it. But the part about "userName" I think it's not being used anywhere since we don't have a user system, so I've removed this part

Reply:
Ei, I still see this code block in home_page.dart . Is that something desired?